homeassistant/components/starlink/binary_sensor.py

```diff
@@ -114,4 +114,9 @@ def is_on(self) -> bool | None:
         entity_category=EntityCategory.DIAGNOSTIC,
         value_fn=lambda data: data.alert["alert_unexpected_location"],
     ),
+    StarlinkBinarySensorEntityDescription(
+        key="connection",
+        device_class=BinarySensorDeviceClass.CONNECTIVITY,
+        value_fn=lambda data: (data.status["state"] == "CONNECTED"),
```

Comment:
```suggestion
        value_fn=lambda data: data.status["state"] == "CONNECTED",
```

What are the other states? Would it make sense to have it be an enum sensor instead?

Reply:
This sensor is specifically binary because the raw state can change with a Starlink software update, so no enum sensor in this PR - [see this thread above for more](https://github.com/home-assistant/core/pull/133184#discussion_r1885105525).

Reply:
I also think we should keep it simple.